### DISCUSSION TITLE
I want to render the home page for the first time server side with default data. But when the search bar is clicked, update the page with the search API response - Using the Next JS App Router

### DISCUSSION THREAD

**Discussion by aypen-git** [vercel/next.js#59015] [App Router]
Here is the scenario -

I have the Home Page - server component which loads with default data.

There is a search Bar component on the home page.

There is a search Bar component on the home page.API and get the response data.

Now, I want to update the home page with the search result instead of the default data.

**Here's the component tree:**

Home - server component
         SearchBar - client component
         CardGrid - server component. Takes data as the prop from the Home Component.

Since Home is a server component, I directly hit the DB. But for search, there is a search API

As per my understanding, I need to use useState to manage search results and pass it as a prop to cardgrid. But Home is a server component.

How to populate data in cardGrid from the server side, but when searched, hit the API to populate the cards.

**Comment by icyJoseph** [comment]
```jsx
export default function Page() {
  const initial = await getDataFromDB()
  // make sure initial is serializable 
  return <ClientComponent initial={initial} />
}
```

> https://react.dev/reference/react/use-client#serializable-types

Then

```jsx
'use client'
import { useState } from 'react';

export const ClientComponent = ({initial}) => {
   const [data, setData] = useState(initial);
   const [isFetching, setIsFetching] = useState(false);

   async function getNewData() {
     // you can instead use swr, or react-query, which can take an initial state as well
     setIsFetching(true);
     const response =  await fetch(/* do asynchronous stuff to get new data */)
     const newData = await response.json() // do error handling
     setData(newData)
     setIsFetching(false);
   }

   return (
    <>
       {/* aria-disabled might be preferred */}
       <button onClick={getNewData} disabled={isFetching}>new data</button>
       <pre>{JSON.stringify(data, null, 2)}</pre>
    </>
   )
}
```
